sklearn/multioutput.py

```diff
@@ -636,22 +637,68 @@ def _check(self):
 
 class _BaseChain(BaseEstimator, metaclass=ABCMeta):
     _parameter_constraints: dict = {
-        "base_estimator": [HasMethods(["fit", "predict"])],
+        "base_estimator": [
+            HasMethods(["fit", "predict"]),
+            StrOptions({"deprecated"}),
+        ],
+        "estimator": [
+            HasMethods(["fit", "predict"]),
+            None,
+        ],
         "order": ["array-like", StrOptions({"random"}), None],
         "cv": ["cv_object", StrOptions({"prefit"})],
         "random_state": ["random_state"],
         "verbose": ["boolean"],
     }
 
+    # TODO(1.8): Remove base_estimator
     def __init__(
-        self, base_estimator, *, order=None, cv=None, random_state=None, verbose=False
+        self,
+        estimator=None,
+        *,
+        order=None,
+        cv=None,
+        random_state=None,
+        verbose=False,
+        base_estimator="deprecated",
     ):
+        self.estimator = estimator
         self.base_estimator = base_estimator
         self.order = order
         self.cv = cv
         self.random_state = random_state
         self.verbose = verbose
 
+    def _validate_estimator(self, default=None):
+        """Check the base estimator.
+
+        Set the `_estimator` attribute.
+        """
+        # TODO(1.8): Remove the self.base_estimator condition when
+        # base_estimator is removed
+        if self.estimator is not None and (
+            self.base_estimator not in [None, "deprecated"]
```

Comment:
Why should we warn in the case of `base_estimator=None`? It seems that our constructor will always set to `"deprecated"` so any other cases mean that our user specifically set `base_estimator` and the warning should be raised.